File: vocabulary-bot-week2/src/main/java/by/iba/bot/vocabulary/config/BotCommand.java
package by.iba.bot.vocabulary.config;

/**
 * This enum contains all commands supported by the Telegram bot
 *
 * @author Mikalai Zaikin (nzaikin@iba.by)
 * @since 4Q2021
 */
public enum BotCommand {
    START("Старт"),
    HELP("Помощь"),
    WORD("Показать новое слово сейчас"),
    ABOUT("Что за бот ?");

    private final String description;

    BotCommand(String description) {
        this.description = description;
    }

    /**
     * Text description of what the command does
     */
    public String getDescription() {
        return description;
    }

    /**
     * Returns a command as it must be typed by the user in the chat, i.e. '/start'
     */
    public String getName() {
        return "/" + name().toLowerCase();
    }
}

File: vocabulary-bot-week3.1/src/main/java/by/iba/bot/vocabulary/command/BotCommand.kt
package by.iba.bot.vocabulary.command

import org.telegram.telegrambots.meta.api.objects.Update

abstract class BotCommand(val command: Command) {

    abstract fun handleUpdate(update: Update, chatId: Long, inputMessageText: String): String

    abstract fun handleUpdate(update: Update, chatId: Long): String

}